src/adam_impact_study/physical_params.py

```diff
@@ -0,0 +1,302 @@
+        while True:
+            chunk = infile.read(65536)
+            if not chunk:
+                break
+            chunk = chunk.replace(b'"', b"")
+            outfile.write(chunk)
+    os.replace(temp_file_path, file_path)
+
+
+def write_phys_params_file(properties_table: pa.Table, properties_file: str) -> None:
+    """
+    Write the physical parameters to a file.
+
+    Parameters
+    ----------
+    properties_table : `pyarrow.Table`
+        Table containing the physical parameters of the impactors.
+    properties_file : str
+        Path to the file where the physical parameters will be saved.
+    """
+    pa.csv.write_csv(
+        properties_table,
+        properties_file,
+        write_options=pa.csv.WriteOptions(
+            include_header=True,
+            delimiter=" ",
+            quoting_style="needed",
+        ),
+    )
+    remove_quotes(properties_file)
+
+
+def select_albedo_from_range(
+    albedo_min: float, albedo_max: float, seed: int = 13612
+) -> float:
+    """
+    Select an albedo from a range.
+
+    Parameters
+    ----------
+    albedo_min : float
+        Minimum albedo value.
+    albedo_max : float
+        Maximum albedo value.
+    seed : int, optional
+        Seed for the random number generator.
+
+    Returns
+    -------
+    albedo : float
+        Albedo value.
+    """
+    rng = np.random.default_rng(seed)
+    return rng.uniform(albedo_min, albedo_max)
+
+
+def select_asteroid_size(min_diam: float, max_diam: float, seed: int = 13612) -> float:
+    """
+    Select an asteroid size from a range.
+
+    Parameters
+    ----------
+    min_diam : float
+        Minimum asteroid diameter.
+    max_diam : float
+        Maximum asteroid diameter.
+    seed : int, optional
+        Seed for the random number generator.
+
+    Returns
+    -------
+    diam : float
+        Asteroid diameter
+    """
+    rng = np.random.default_rng(seed)
+    return rng.uniform(min_diam, max_diam)
+
+
+def determine_ast_class(percent_C: float, percent_S: float) -> str:
+    """
+    Determine the asteroid class based on the percentage of C and S asteroids.
+
+    Parameters
+    ----------
+    percent_C : float
+        Percentage of C asteroids.
+    percent_S : float
+        Percentage of S asteroids.
+
+    Returns
+    -------
+    ast_class : str
+        Asteroid class.
+    """
+    assert percent_C + percent_S == 1, "Percentage of C and S asteroids must equal 1"
+    rng = np.random.default_rng()
+    return "C" if rng.random() < percent_C else "S"
+
+
+def calculate_H(diameter: float, albedo: float) -> float:
+    """
+    Calculate the absolute magnitude of an asteroid.
+
+    Parameters
+    ----------
+    diameter : float
+        Asteroid diameter.
+    albedo : float
+        Asteroid albedo.
+
+    Returns
+    -------
+    H : float
+        Absolute magnitude.
+    """
+    return 15.618 - 5 * np.log10(diameter) - 2.5 * np.log10(albedo)
```

Comment:
Can confirm this is equivalent to the alternate format:
H = 5 log (1329 km / (d * sqrt(pv)))